src/storage/btree.h

```diff
@@ -532,6 +534,123 @@ struct btree_object_info
 #define BTREE_OBJECT_INFO_INITIALIZER \
   { OID_INITIALIZER, OID_INITIALIZER, BTREE_MVCC_INFO_INITIALIZER }
 
+struct key_oid
+{
+  DB_VALUE m_key;
+  OID m_oid;
+};
+
+// *INDENT-OFF*
+
+#if !defined (NDEBUG)
+struct advance_page_trace
+{
+  typedef enum 
+    { 
+      ONE_KEY = 0,
+      EQ_KEY,
+      LT_KEY,
+      GT_KEY
+    } advance_case;
+
+  advance_case m_adv_case;
+  int m_key_cnt;
+
+  VPID m_curr_page_vpid;
+  PAGE_PTR m_curr_page_ptr;
+
+  INT16 m_slot_id;
+  VPID m_child_vpid;
+
+  advance_page_trace (advance_case c, int key_cnt, VPID curr_vpid, PAGE_PTR curr_page_ptr, INT16 slot_id,
+                      VPID child_vpid)
+    : m_adv_case (c)
+    , m_key_cnt (key_cnt)
+    , m_curr_page_vpid (curr_vpid)
+    , m_curr_page_ptr (curr_page_ptr)
+    , m_slot_id (slot_id)
+    , m_child_vpid (child_vpid)
+    {
+    }
+};
+#endif
+
+struct page_key_boundary
+{
+  DB_VALUE m_left_key;
+  DB_VALUE m_right_key;
+
+  bool m_is_inf_left_key;
+  bool m_is_inf_right_key;
+
+  page_key_boundary ();
+  ~page_key_boundary ();
+
+#if !defined (NDEBUG)
+  std::vector <advance_page_trace> m_trace;
+  void add_trace (advance_page_trace &&trace_point)
+    {
+      m_trace.push_back (trace_point);
+    }
+#endif
+};
+
+struct btree_insert_list
```

Comment:
can we expose here just a class that collects the key_oid's to try to keep header clean? one possibility is to move page_key_boundary to insert_helper or to another structure static to btree.c.

Reply:
The next best candidate structure would be insert_helper; however this requires an initializer macro, we would need an initializer for DB_VALUE for instance. I don't think we want to keep such constructs in the code.
I think is better to have this change into a future PR where insert_helper is a struct with proper constructor and does not require initializer.